src/logion/lib/db/collections.ts

```diff
@@ -0,0 +1,8 @@
+export interface HasIndex {
+    index?: number;
+}
+
+export function list<T>(items: HasIndex[], mapper: (item: any) => T): Array<T> {
```

Comment:
Because the goal here is to rebuid a list based on entities stored in a table. The fact that we need to sort and map is just the way to achieve the purpose.